keyboards/evyd13/atom47/keymaps/phsc138/keymap.c

```diff
@@ -0,0 +1,390 @@
+static tap pn_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void pn_finished(qk_tap_dance_state_t *state, void *user_data) {
+    pn_tap_state.state = cur_dance(state);
+    switch(pn_tap_state.state) {
+        case SINGLE_TAP: layer_on(_PROG); break;
+        case SINGLE_HOLD: layer_on(_NUM); break;
+        //case DOUBLE_TAP: register_code(KC_ESC); break;
+        //case DOUBLE_HOLD: register_code(KC_LALT); break;
+        //case DOUBLE_SINGLE_TAP: register_code(KC_X); unregister_code(KC_X); register_code(KC_X);
+        //Last case is for fast typing. Assuming your key is `f`:
+        //For example, when typing the word `buffer`, and you want to make sure that you send `ff` and not `Esc`.
+        //In order to type `ff` when typing fast, the next character will have to be hit within the `TAPPING_TERM`, which by default is 200ms.
+    }
+}
+
+void pn_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(pn_tap_state.state) {
+        //case SINGLE_TAP: unregister_code(KC_X); break;
+        case SINGLE_HOLD: layer_off(_NUM); break;
+        //case DOUBLE_TAP: unregister_code(KC_ESC); break;
+        //case DOUBLE_HOLD: unregister_code(KC_LALT);
+        //case DOUBLE_SINGLE_TAP: unregister_code(KC_X);
+    }
+    pn_tap_state.state = 0;
+}
+
+static tap lalt_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void lalt_finished(qk_tap_dance_state_t *state, void *user_data) {
+    lalt_tap_state.state = cur_dance(state);
+    switch(lalt_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_LBRC); break;
+        case SINGLE_HOLD: register_code(KC_LALT); break;
+    }
+}
+
+void lalt_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(lalt_tap_state.state) {
+        case SINGLE_TAP: unregister_code(KC_LBRC); break;
+        case SINGLE_HOLD: unregister_code(KC_LALT); break;
+    }
+    lalt_tap_state.state = 0;
+}
+
+static tap ralt_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void ralt_finished(qk_tap_dance_state_t *state, void *user_data) {
+    ralt_tap_state.state = cur_dance(state);
+    switch(ralt_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_RBRC); break;
+        case SINGLE_HOLD: register_code(KC_RALT); break;
+    }
+}
+
+void ralt_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(ralt_tap_state.state) {
+        case SINGLE_TAP: unregister_code(KC_RBRC); break;
+        case SINGLE_HOLD: unregister_code(KC_RALT); break;
+    }
+    ralt_tap_state.state = 0;
+}
+
+static tap rctl_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void rctl_finished(qk_tap_dance_state_t *state, void *user_data) {
+    rctl_tap_state.state = cur_dance(state);
+    switch(rctl_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_RSFT); register_code(KC_RBRC); break;
+        case SINGLE_HOLD: register_code(KC_RCTL); break;
+    }
+}
+
+void rctl_reset(qk_tap_dance_state_t *state, void *user_data) {
```

Comment:
```suggestion
void rctl_reset(tap_dance_state_t *state, void *user_data) {
```

Reply:
Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)